src/wallet/api/wallet.h

```diff
@@ -151,6 +152,10 @@ class WalletImpl : public Wallet
     bool                m_recoveringFromSeed;
     std::atomic<bool>   m_synchronized;
     bool                m_rebuildWalletCache;
+    // cache connection status to avoid unnecessary RPC calls
+    mutable bool        m_is_connected;
+    // on asynchronous wallet close we need to make sure no calls are made after shutdown started.
+    bool                m_is_shutting_down;
```

Comment:
changed to atomic bool. 
removed m_is_shutting_down btw. We handle that client side instead.